src/Compilers/CSharp/Portable/Symbols/Metadata/PE/SymbolFactory.cs

```diff
@@ -153,7 +154,12 @@ internal override TypeSymbol GetUnsupportedMetadataTypeSymbol(PEModuleSymbol mod
                 return new UnsupportedMetadataTypeSymbol();
             }
 
-            TypeMap substitution = new TypeMap(typeParameters, arguments);
+            if (System.Linq.Enumerable.Any(arguments, arg => !arg.Value.IsDefaultOrEmpty))
```

Comment:
This code is going to be replaced with a real handling of custom modifiers in the next change.